contracts/actions/liquity/bonds/FetchBondId.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.10;
+
+import "../../ActionBase.sol";
+import "../../../core/strategy/SubStorage.sol";
+import "../../../core/strategy/StrategyModel.sol";
+import "../../../actions/liquity/helpers/CBHelper.sol";
+
+/// @title Special action to fetch BondId for the Liquity payback from CB strategy and to deactivate rebond strategy if bond from rebond strat was used
+contract FetchBondId is ActionBase, CBHelper {
+
+    error WrongSourceType(SourceType);
+    error SubDatHashMismatch(uint256, bytes32, bytes32);
+
+    enum SourceType{
+        BOND,
+        SUB
+    }
+
+    /// @param paybackSourceId Id of the payback source, can be either bondId or rebond strat subId
+    /// @param sourceType number indicating if sourceId refers to a bondId or subId
+    /// @param cbRebondBondId Id of the current bond in the Rebond sub (only used if paybackSourceId is of a sub, otherwise 0)
+    struct Params {
```

Comment:
Naming is to generalized `paybackSourceId` gives 0 context of what it does and the action is very specific to this strategy, for example what I consider more descriptive names

```/// @param bondIdOrSubId Id of the bond or sub
    /// @param isSub If the bondIdOrSubId is a sub or not
    /// @param bondIdFromSub BondId from the sub if isSub is true
    struct Params {
        uint256 bondIdOrSubId;
        bool isSub;
        uint256 bondIdFromSub;
    }```